test/stub-test.js

```diff
@@ -1271,34 +1306,41 @@ describe("stub", function () {
 
         it("plays nice with throws", function () {
             var stub = createStub().throws().yields();
-            var spy = createSpy();
+            var callback = createStub().returns("return value");
             assert.exception(function () {
-                stub(spy);
+                stub(callback);
             });
-            assert(spy.calledOnce);
+            assert(callback.calledOnce);
         });
 
         it("plays nice with returns", function () {
             var obj = {};
             var stub = createStub().returns(obj).yields();
-            var spy = createSpy();
-            assert.same(stub(spy), obj);
-            assert(spy.calledOnce);
+            var callback = createStub().returns("return value");
+            assert.same(stub(callback), obj);
+            assert(callback.calledOnce);
         });
 
         it("plays nice with returnsArg", function () {
             var stub = createStub().returnsArg(0).yields();
-            var spy = createSpy();
-            assert.same(stub(spy), spy);
-            assert(spy.calledOnce);
+            var callback = createStub().returns("return value");
+            assert.same(stub(callback), callback);
+            assert(callback.calledOnce);
         });
 
         it("plays nice with returnsThis", function () {
             var obj = {};
             var stub = createStub().returnsThis().yields();
-            var spy = createSpy();
-            assert.same(stub.call(obj, spy), obj);
-            assert(spy.calledOnce);
+            var callback = createStub().returns("return value");
+            assert.same(stub.call(obj, callback), obj);
+            assert(callback.calledOnce);
+        });
+
+        it("returns the result of the yielded callback", function () {
+            var stub = createStub().yields();
+            var callback = createStub().returns("return value");
+            assert.same(stub(callback), "return value");
```

Comment:
Minor styling: Can you separate setup / testing code from assertions with a blank line? This style should be used everywhere. You did it right in the first few tests already. It makes the tests easier to read :)